### PR TITLE
Upgrade DetailsComponent to Slots V2

### PR DESCRIPTION
**Pull request by srt32**
Previously

```
<%= render(Primer::DetailsComponent.new(overlay: :default, reset: true, position: :relative)) do |c| %>
  <% c.slot(:summary) do %>
    Dropdown
  <% end %>

  <% c.slot(:body) do %>
  <% end %>
<% end %>
```

and now

```
<%= render(Primer::DetailsComponent.new(overlay: :default, reset: true, position: :relative)) do |c| %>
  <% c.summary do %>
    Dropdown
  <% end %>

  <% c.body do %>
   Body
  <% end %>
<% end %>
```

### CODE REVIEW

File: app/components/primer/details_component.rb [hunk 1]
```diff
@@ -12,9 +12,42 @@ class DetailsComponent < Primer::Component
       :dark => "details-overlay details-overlay-dark"
     }.freeze
 
-    with_slot :body, class_name: "Body"
-    with_slot :summary, class_name: "Summary"
+    renders_one :body, class_name: "Body"
```
Comment (manuelpuyol) [inline app/components/primer/details_component.rb:15]:
```suggestion
    renders_one :body, "Body"
```

File: app/components/primer/details_component.rb [hunk 2]
```diff
@@ -12,9 +12,42 @@ class DetailsComponent < Primer::Component
...
+    renders_one :summary, class_name: "Summary"
```
Comment (manuelpuyol) [inline app/components/primer/details_component.rb:16]:
```suggestion
    renders_one :body, "Body"
    renders_one :summary, "Summary"
```

Comment (srt32) [reply app/components/primer/details_component.rb:16]: I'm still seeing the same problem. I'm gunna convert to the lamba format. Still not sure what's going wrong though...

**Comment by srt32** [comment]
I forgot we had the issue with safari with the examples...

image

File: app/components/primer/details_component.rb [hunk 3]
```diff
@@ -12,9 +12,60 @@ class DetailsComponent < Primer::Component
...
+    #   <%= render(Primer::DetailsComponent.new(overlay: :default, reset: true, position: :relative)) do |c| %>
+    #     <% c.summary(button: false) do %>
+    #       Click me
+    #     <% end %>
+    #
+    #     <% c.body do %>
+    #       Body
+    #     <% end %>
+    #   <% end %>
+    #
```
Comment (srt32) [inline app/components/primer/details_component.rb:68]:
```suggestion
```

remove examples as safari display is still broken.

File: app/components/primer/details_component.rb [hunk 4]
```diff
@@ -12,8 +12,26 @@ class DetailsComponent < Primer::Component
       :dark => "details-overlay details-overlay-dark"
     }.freeze
 
-    with_slot :body, class_name: "Body"
-    with_slot :summary, class_name: "Summary"
+    # Use the Summary slot as a trigger to reveal the content.
+    #
+    # @param kwargs [Hash] The same arguments as <%= link_to_system_arguments_docs %>.
```
Comment (manuelpuyol) [inline app/components/primer/details_component.rb:17]: missing `button` param

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,10 @@
 
 ## main
 
+* **Breaking change**: Upgrade `DetailsComponent` to use Slots V2.
```
Comment (manuelpuyol) [inline CHANGELOG.md:5]: might we put this at the bottom to group all `breaking changes`?

File: app/components/primer/details_component.rb (hunk 4 above)
Comment (srt32) [inline app/components/primer/details_component.rb:17]:
```suggestion
    # @param button [Boolean] Whether to render the Summary as a button or not.
    # @param kwargs [Hash] The same arguments as <%= link_to_system_arguments_docs %>.
```

**Review by manuelpuyol — APPROVED** [review]

**Pull request merged**